reflex/state.py

```diff
@@ -1238,6 +1457,28 @@ def _dirty_computed_vars(self, from_vars: set[str] | None = None) -> set[str]:
             for cvar in self._computed_var_dependencies[dirty_var]
         )
 
+    @classmethod
+    def _potentially_dirty_substates(cls) -> set[Type[BaseState]]:
```

Comment:
So does this mean any substate with a computed var will always be refetched on each event?

Reply:
yes that's a correct read.

currently we promise that computed vars will always update and cached_vars will update when their deps change. to keep that promise, we have to fetch the states containing those vars.

it's not really "nice", but it's better than getting all the states all the time, and users can optimize their own apps by limiting the use of computed vars or replacing them with cached vars in a shallow hierarchy.

this keeps the mental model simple for downstream users.

Reply:
sounds good - I suppose in reflex-web we don't have many of these which is good. We should document all this thoroughly later in the substate section